src/payment/strategies/credit-card/credit-card-cardinal-payment-strategy.ts

```diff
@@ -17,15 +17,15 @@ import PaymentStrategy from '../payment-strategy';
 
 import { CardinalClient, CardinalOrderData } from './index';
 
-export default class CyberSourcePaymentStrategy implements PaymentStrategy {
-    private _paymentMethod?: PaymentMethod;
+export default class CreditCardCardinalPaymentStrategy implements PaymentStrategy {
```

Comment:
I think a better abstraction is to use composition rather than inheritance. Now you're making `PaypalProPaymentStrategy` inheriting a strategy that's supposed to be a strategy built specifically for CyberSource. We usually only use inheritance when a sub-class is a type of the parent class. In this case, it's not the right abstraction to use. You can tell by looking at the `execute` method of `PaypalProPaymentStrategy`, where half of the method is overridden because PayPal requires a different behaviour. 

I think what you really want is to share code between the two strategies - you want them to have the capability of triggering 3DS flow using the Cardinal client. You can do that by simply moving the shared code related to Cardinal into a separate class and inject instances of that class into the aforementioned strategies.